src/main/java/com/baidu/hugegraph/manager/BackupRestoreBaseManager.java

```diff
@@ -0,0 +1,247 @@
+        } catch (Exception e) {
+            throw new ToolsException("Failed to serialize %s to %s",
+                                     e, type, path);
+        }
+    }
+
+    protected void read(String file, HugeType type,
+                        BiConsumer<String, String> consumer) {
+        InputStream is = this.inputStream(file);
+        try (InputStreamReader isr = new InputStreamReader(is, API.CHARSET);
+             BufferedReader reader = new BufferedReader(isr)) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                consumer.accept(type.string(), line);
+            }
+        } catch (IOException e) {
+            throw new ToolsException("Failed to deserialize %s from %s",
+                                     e, type, file);
+        }
+    }
+
+    private OutputStream outputStream(String file) {
+        OutputStream os;
+        if (this.outputStreams.get(file) == null) {
+            os = this.directory.outputStream(file, true);
+            this.outputStreams.putIfAbsent(file, os);
+        }
+        return this.outputStreams.get(file);
+    }
+
+    private InputStream inputStream(String file) {
+        InputStream is;
+        if (this.inputStreams.get(file) == null) {
+            is = this.directory.inputStream(file);
+            this.inputStreams.putIfAbsent(file, is);
+        }
+        return this.inputStreams.get(file);
+    }
+
+    protected List<String> filesWithPrefix(HugeType type) {
+        List<String> files = new ArrayList<>();
+        for (String file : this.directory.files()) {
+            if (file.startsWith(type.string())) {
+                files.add(file);
+            }
+        }
+        return files;
+    }
+
+    public void startTimer() {
+        this.startTime = System.currentTimeMillis();
+    }
+
+    public long elapseSeconds() {
+        E.checkState(this.startTime != 0,
+                     "Must call startTimer() to set start time, " +
+                     "before call elapse()");
+        return (System.currentTimeMillis() - this.startTime) / 1000;
+    }
+
+    protected void printSummary() {
+        this.printSummary(this.type());
+    }
+
+    protected void printSummary(String type) {
+        Map<String, Long> summary = ImmutableMap.<String, Long>builder()
+                .put("property key number", this.propertyKeyCounter.longValue())
+                .put("vertex label number", this.vertexLabelCounter.longValue())
+                .put("edge label number", this.edgeLabelCounter.longValue())
+                .put("index label number", this.indexLabelCounter.longValue())
+                .put("vertex number", this.vertexCounter.longValue())
+                .put("edge number", this.edgeCounter.longValue()).build();
+        Printer.printMap(type + " summary", summary);
+
+        Printer.printKV("cost time(s)", this.elapseSeconds());
+    }
+
+    protected static void ensureDirectoryExist(String dir) {
```

Comment:
unused method?

Reply:
used by ensure log dir in local fs exist

Reply:
ok, it can be moved to LocalDirectory or Util